bom-aware-scanner.go

```diff
@@ -0,0 +1,35 @@
+package astisub
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+)
+
+type BomAwareScanner struct {
+	*bufio.Scanner
+	lookingForBom bool
+}
+
+func NewBomAwareScanner(r io.Reader) *BomAwareScanner {
```

Comment:
Can you guide me through why you need to create a specific scanner instead of just trimming the BOM byte on the first line? Is there any gain with this scanner?

Reply:
I thought it could be reused for other formats. For instance, WebVTT. But maybe I'm indeed over-engineering.

Reply:
Also, do you plan to make go-astisub work for other encodings, like UTF-16? I think a dedicated scanner could abstract away their differences, including different BOMs.

Reply:
Making go-astisub compatible with UTF-16 is a nice idea indeed. But in the context of this PR, I'd rather keep things simple and only trim BOM bytes.

Reply:
Could you do it yourself? I can't come up with a nice implementation, don't want to duplicate stuff.

Reply:
Sure I can do once the PR is merged.

You'll need to remove your new scanner though.

Reply:
Ok, I have now removed it.